includes/REST_API/Publisher_Logos_Controller.php

```diff
@@ -0,0 +1,435 @@
+		$results         = [];
+
+		foreach ( $publisher_logos as $logo ) {
+			/**
+			 * We're expecting a post object after the filtering above.
+			 *
+			 * @var WP_Post $post
+			 */
+			$post = get_post( $logo );
+
+			$data      = $this->prepare_item_for_response( $post, $request );
+			$results[] = $this->prepare_response_for_collection( $data );
+		}
+
+		return rest_ensure_response( $results );
+	}
+
+	/**
+	 * Adds a new publisher logo to the collection.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function create_item( $request ) {
+		$publisher_logos = $this->filter_publisher_logos( (array) get_option( Settings::SETTING_NAME_PUBLISHER_LOGOS, [] ) );
+
+		$post = get_post( $request['id'] );
+
+		if ( ! $post || 'attachment' !== $post->post_type ) {
+			return new WP_Error(
+				'rest_invalid_id',
+				__( 'Invalid ID', 'web-stories' ),
+				[ 'status' => 400 ]
+			);
+		}
+
+		if ( in_array( $post->ID, $publisher_logos, true ) ) {
+			return new WP_Error(
+				'rest_publisher_logo_exists',
+				__( 'Publisher logo already exists', 'web-stories' ),
+				[ 'status' => 400 ]
+			);
+		}
+
+		$publisher_logos[] = $post->ID;
+
+		update_option( Settings::SETTING_NAME_PUBLISHER_LOGOS, $publisher_logos );
+
+		$active_publisher_logo_id = absint( get_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO ) );
+
+		if ( 1 === count( $publisher_logos ) || ! in_array( $active_publisher_logo_id, $publisher_logos, true ) ) {
+			update_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO, $post->ID );
+		}
+
+		return $this->prepare_item_for_response( $post, $request );
+	}
+
+	/**
+	 * Removes a publisher logo from the collection.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function delete_item( $request ) {
+		$post = $this->get_publisher_logo( $request['id'] );
+
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		$publisher_logos = $this->filter_publisher_logos( (array) get_option( Settings::SETTING_NAME_PUBLISHER_LOGOS, [] ) );
+		$publisher_logos = array_values( array_diff( $publisher_logos, [ $post->ID ] ) );
+
+		$active_publisher_logo_id = absint( get_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO ) );
+
+		if ( $post->ID === $active_publisher_logo_id || ! in_array( $active_publisher_logo_id, $publisher_logos, true ) ) {
+			// Mark the first available publisher logo as the new default.
+			update_option( Settings::SETTING_NAME_ACTIVE_PUBLISHER_LOGO, ! empty( $publisher_logos[0] ) ? $publisher_logos[0] : 0 );
+		}
+
+		update_option( Settings::SETTING_NAME_PUBLISHER_LOGOS, $publisher_logos );
+
+		return $this->prepare_item_for_response( $post, $request );
```

Comment:
Delete endpoints response normally like [this](https://github.com/WordPress/wordpress-develop/blob/cb5e2611a618e8de608469c5d56a6569cb6a1a01/src/wp-includes/rest-api/endpoints/class-wp-rest-widgets-controller.php#L387-L392).